dotCMS/src/integration-test/java/com/dotcms/security/apps/AppsAPIImplTest.java

```diff
@@ -1021,4 +1024,55 @@ public void Test_File_Comparison_Is_Case_Sensitive()
         //The one from the system-folder takes precedence.
         assertEquals("system-app", impl.getDescription());
     }
+
+    /**
+     * Given Scenario: This test creates secrets then exports them Then re-imports the file with the generated secrets.
+     * Expected Result: The test should be Able to recreate the secrets from the generated file.
+     * @throws DotDataException
+     * @throws DotSecurityException
+     * @throws IOException
+     * @throws EncryptorException
+     * @throws ClassNotFoundException
+     */
+    @Test
+    public void Test_Create_Secrets_Then_Export_Them_Then_Import_Then_Save()
+            throws DotDataException, DotSecurityException, IOException, EncryptorException, ClassNotFoundException {
+        final User admin = TestUserUtils.getAdminUser();
+        final Host site = new SiteDataGen().nextPersisted();
+        final AppsAPI api = APILocator.getAppsAPI();
+        final AppSecrets.Builder builder1 = new AppSecrets.Builder();
+        final String appKey = "appKey-1-Host-1";
+        final AppSecrets secrets = builder1.withKey(appKey)
+                .withHiddenSecret("test:secret1", "secret-1")
+                .withHiddenSecret("test:secret2", "secret-2")
+                .withHiddenSecret("test:secret3", "secret3")
+                .withHiddenSecret("test:secret4", "secret-4")
+                .build();
+        //Save it
+        api.saveSecrets(secrets, site, admin);
+        final Optional<AppSecrets> secretsOptional = api.getSecrets(appKey, site, admin);
+        final AppSecrets appSecretsPostSave = secretsOptional.get();
+
+        //AES only supports key sizes of 16, 24 or 32 bytes.
+        final String password = RandomStringUtils.randomAlphanumeric(32);
+        final Key key = AppsUtil.generateKey(password);
+        final File exportSecretsFile = api.exportSecrets(key, true, null, admin);
+        assertTrue(exportSecretsFile.exists());
+
+        //Remove so we can re import them.
+        api.deleteSecrets(appKey, site, admin);
+
+        final Map<String, List<AppSecrets>> secretAppsBySiteId = api.importSecrets(exportSecretsFile.toPath(), key, admin);
+        assertTrue(secretAppsBySiteId.size() >= 1);
```

Comment:
Codacy found an issue: [Substitute calls to size() == 0 (or size() != 0, size() > 0, size() < 1) with calls to isEmpty()](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5931733)